request_handler.h

```diff
@@ -14,12 +15,47 @@ class RequestHandler {
 		ERROR = 1
 	};
 
-	// requestData is the full http request
-	// request is the parsed header
+	// Initializes the handler. Returns a response code indicating success or
+	// failure condition.
+	// uri_prefix is the value in the config file that this handler will run for.
+	// config is the contents of the child block for this handler ONLY.
+	virtual Status Init(const std::string& uri_prefix, const NginxConfig& config) = 0;
+
+	// Creation Magic by name of handler as specified in config
+	static RequestHandler* CreateByName(const std::string& type);
+
+	// Handles an HTTP request, and generates a response. Returns a response code
+	// indicating success or failure condition. If ResponseCode is not OK, the
+	// contents of the response object are undefined, and the server will return
+	// HTTP code 500.
 	virtual RequestHandler::Status HandleRequest(const Request& request, Response* response) const = 0;
 };
 
 // map of k,v pairs, where k = path and v = pointer to request handler
-typedef std::map<const std::string, std::unique_ptr<RequestHandler>> HandlerContainer;
+typedef std::map<const std::string, RequestHandler*> HandlerContainer;
```

Comment:
Changed to shared_ptr. CreateByName does not matter for common API, so changed to return shared_ptr as well.